### DISCUSSION TITLE
Setting a component's variable to localStorage / global variable value on the initial render

### DISCUSSION THREAD

**Discussion by ZachSaucier** [sveltejs/kit#11137] [Q&A] [answered]
I'm using Svelte + SvelteKit.

I'd like to set _the initial_ value of a variable in my component based on the user's previously saved value in `localStorage`. I can get it to load _after_ the initial load using:

```
let foo = false;

if (browser) {
  foo = localStorage.collapsed === 'true'
}
```

This seems to run once the page has loaded, which in my case means there's a flash of the initial value.

I also know I can use `<svelte:head>` to get the initial `localStorage` value before the page renders:

```
<svelte:head>
  <script>
    if (localStorage.foo === 'true') {
      window.foo = true;
    }
  </script>
</svelte:head>
```

But how can I update my component's value based on this global variable / what's obtained in the `<head>`?

For CSS, I can attach a class to the `documentElement` or `body` but that doesn't help me here.

Repl: https://svelte.dev/repl/3e4099fc70814541ae837ff7da50926a?version=4.2.7

**Comment by david-plugge** [comment, marked as answer]
If you are using ssr (which is on by default) the code is executed on the server where the clients localStorage is not available. Therefore you simply can't achieve it.

If you need ssr support you have to use another state method like a cookie or the url. Make sure to read the state in a load function.

**Reply by ZachSaucier** [reply]
Thanks for the reply! I am using SSR.

To confirm: even though I can use `<svelte:head>` to get the variable value, there's no way to pass that to the component for the initial render?

**Reply by david-plugge** [reply]
The html is generated on the server and localStorage is only available on the client. The generated HTML gets send to your browser which then displays it. Once the browser has loaded all the js files, it will "take over" and will be able to provide the data inside localStorage. The moment when the html from the server is rendered, 'foo' will be your default value of false and once the client has loaded, it will be whatever is inside localStorage.

So in short: yes, you can NOT use localStorage for your use case

**Reply by ZachSaucier** [reply]
Thanks for the full explanation!
